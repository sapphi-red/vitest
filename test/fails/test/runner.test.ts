import { resolve } from 'pathe'
import fg from 'fast-glob'
import { execaSync } from 'execa'
import { describe, expect, it } from 'vitest'
import { normalizePath } from 'vite'

describe('should fails', async() => {
  const root = resolve(__dirname, '../fixtures')
  const files = await fg('*.test.ts', { cwd: root })

  for (const file of files) {
    it(file, async() => {
      let error: any
      try {
        execaSync('npx', ['vitest', 'run', file], {
          cwd: root,
          env: {
            ...process.env,
            CI: 'true',
            NO_COLOR: 'true',
          },
        })
      }
      catch (e) {
        error = e
      }

      expect(error).toBeTruthy()
      const msg = String(error)
        .split(/\n/g)
        .reverse()
        .find(i => i.includes('Error: '))
        ?.trim().replace(normalizePath(root), '')
      expect(msg).toMatchSnapshot(file)
    }, 10000)
  }
})
