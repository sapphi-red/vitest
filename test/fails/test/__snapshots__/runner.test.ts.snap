// Vitest Snapshot v1

exports[`should fails > empty.test.ts > empty.test.ts 1`] = `"Error: No test suite found in file /empty.test.ts"`;

exports[`should fails > expect.test.ts > expect.test.ts 1`] = `"AssertionError: expected 2 to deeply equal 3"`;

exports[`should fails > hook-timeout.test.ts > hook-timeout.test.ts 1`] = `"Error: Hook timed out in 10ms."`;

exports[`should fails > nested-suite.test.ts > nested-suite.test.ts 1`] = `"AssertionError: expected true to be false // Object.is equality"`;

exports[`should fails > stall.test.ts > stall.test.ts 1`] = `"TypeError: failure"`;

exports[`should fails > timeout.test.ts > timeout.test.ts 1`] = `"Error: Test timed out in 10ms."`;
